### ISSUE TITLE
[Story] Send pin-drop location

### ISSUE THREAD

**Issue by Johennes** [vector-im/element-meta#1674]
* As a user
* I want to send pin drop locations
* so that I can chat with others about locations different from my own

# Size estimate

L

# Dependencies

* https://github.com/vector-im/element-meta/issues/1684

# Acceptance criteria

* The button to initiate location sending is placed in the composer menu
* Tapping the button brings up the sending map view
* Default map center coordinate (latitude: 49.843, longitude: 9.902056)
* Default zoom level = 2.7
* The map view shows the map provider's copyright attribution
* The map view can be panned and zoomed
* The map view can't be rotated
* Once confirmed, the location event is sent
* `uri` contains the coordinates
* `m.asset` is set to `m.pin`
* `description` is null/empty (i.e. we don't do any geocoding for now)
* `zoom_level` inside `m.location` is set to 15
* `body` text format:
    - Example: `Location was shared at geo:51.50998,-0.133700000000033;u=5 as of 2023-07-18 09:23:20 +0000` **(TBC)**
    - `m.text` will is the same of the body (the rust SDK is already doing it)
* Geo URI
    - Include the uncertainty parameter eg `u=20` only when sending the user's current location
* Analytics tracking
  * When showing the sending map view
    *  Event: [`$screen`](https://github.com/matrix-org/matrix-analytics-events/blob/main/schemas/MobileScreen.json)
    * Property: `screenName = LocationSend`
  * When sending a location
    * Event: [`Composer`](https://github.com/matrix-org/matrix-analytics-events/blob/main/schemas/Composer.json)
    * Property: `isLocation = true`
    * Property: `locationType = PinDrop`
    * Other existing properties on the event are set according to context
    * When sending other events we set `isLocation = false` and leave `locationType` empty

# Out of scope

* Send current location & location button – https://github.com/vector-im/element-meta/issues/1682
* Rendering location events – https://github.com/vector-im/element-meta/issues/1675

### Android
- [x] Design sign-off on completion
- [x] QA sign-off on completion
- [x] Product sign-off on completion

### iOS
- [x] Design sign-off on completion
- [x] QA sign-off on completion
- [x] Product sign-off on completion

# Subtasks

```[tasklist]
### Android
- [ ] https://github.com/vector-im/element-x-android/issues/690
- [ ] https://github.com/vector-im/element-x-android/issues/709
```

```[tasklist]
### iOS
- [ ] https://github.com/vector-im/element-x-ios/issues/1116
- [ ] https://github.com/vector-im/element-x-ios/pull/1241
- [ ] https://github.com/vector-im/element-x-ios/pull/1354
```

```[tasklist]
### Other
- [ ] https://github.com/matrix-org/matrix-rust-sdk/pull/2187
```

**Comment by alfogrillo** [comment]
@julioromano @flescio can you please edit the issue resolving all the **TBD** where we took a decision already?

**Comment by Johennes** [comment]
It looks like the TBDs are still not resolved in the description. @alfogrillo would you mind taking this story over on the iOS side and clarifying things with @julioromano? We may not have to do any work and just need to document what we've implemented. Once that is done, you can move this story into "Reviewed" on both platforms.

**Comment by alfogrillo** [comment]
>  @alfogrillo would you mind taking this story over on the iOS side and clarifying things with @julioromano? 

I resolved all the TBDs expect the format of the location event `body`.
(we need to take a decision for it)
